### DISCUSSION TITLE
Why WrapDB over an existing tool like Conan?

### DISCUSSION THREAD

**Discussion by wbehrens-on-gh** [mesonbuild/meson#10671] [Q&A]
I was wondering what the reason/advantage of creating and maintaining WrapDB rather then using an existing technology like Conan (which is also written in Python) to manage C/C++ dependencies?

**Comment by eli-schwartz** [comment]
The WrapDB isn't limited to C/C++ dependencies, for example it can be used to provide custom_target program utilities or e.g. wayland-protocols XML data inputs.

It also provides seamless on-demand fallback.

You can use it to do full-application LTO.

Also, it's built in, because Meson needed to provide subproject support either way.

...

That being said, Meson is happy to use conan-provided dependencies as "system" versions.
